// @atlaskit/app-provider/dist/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function get() {
    return _themeProvider.default;
  }
});
Object.defineProperty(exports, "UNSAFE_useColorModeForMigration", {
  enumerable: true,
  get: function get() {
    return _themeProvider.UNSAFE_useColorModeForMigration;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _appProvider.default;
  }
});
Object.defineProperty(exports, "useColorMode", {
  enumerable: true,
  get: function get() {
    return _themeProvider.useColorMode;
  }
});
Object.defineProperty(exports, "useRouterLink", {
  enumerable: true,
  get: function get() {
    return _useRouterLink.default;
  }
});
Object.defineProperty(exports, "useSetColorMode", {
  enumerable: true,
  get: function get() {
    return _themeProvider.useSetColorMode;
  }
});
Object.defineProperty(exports, "useSetTheme", {
  enumerable: true,
  get: function get() {
    return _themeProvider.useSetTheme;
  }
});
Object.defineProperty(exports, "useTheme", {
  enumerable: true,
  get: function get() {
    return _themeProvider.useTheme;
  }
});
var _appProvider = _interopRequireDefault(require("./app-provider"));
var _themeProvider = _interopRequireWildcard(require("./theme-provider"));
var _useRouterLink = _interopRequireDefault(require("./router-link-provider/hooks/use-router-link"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
