// @atlaskit/table/dist/cjs/head-cell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _primitives = require("@atlaskit/primitives");
var _th = require("./ui/th");
// eslint-disable-next-line @atlaskit/design-system/no-emotion-primitives -- to be migrated to @atlaskit/primitives/compiled – go/akcss

/**
 * __HeadCell__
 *
 * HeadCell element
 */
var HeadCell = function HeadCell(_ref) {
  var children = _ref.children,
    align = _ref.align,
    testId = _ref.testId,
    backgroundColor = _ref.backgroundColor,
    _ref$scope = _ref.scope,
    scope = _ref$scope === void 0 ? 'col' : _ref$scope;
  return /*#__PURE__*/_react.default.createElement(_th.TH, {
    scope: scope,
    align: align,
    testId: testId,
    backgroundColor: backgroundColor
  }, children && /*#__PURE__*/_react.default.createElement(_primitives.Text, {
    weight: "medium"
  }, children));
};
var _default = exports.default = HeadCell;
