// @atlaskit/table/dist/cjs/sortable-column.js
/* sortable-column.tsx generated by @compiled/babel-plugin v0.38.1 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
require("./sortable-column.compiled.css");
var _react = _interopRequireWildcard(require("react"));
var React = _react;
var _runtime = require("@compiled/react/runtime");
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _button = _interopRequireDefault(require("@atlaskit/button"));
var _tooltip = _interopRequireDefault(require("@atlaskit/tooltip"));
var _useTable2 = require("./hooks/use-table");
var _sortIcon = require("./ui/sort-icon");
var _th = require("./ui/th");
var _excluded = ["name", "testId", "onClick", "children"];
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
/**
 * TODO these need to be i18n supported
 */
var sortingMessages = {
  unsorted: {
    string: 'Sort from A to Z',
    number: 'Sort from 0 to 9'
  },
  ascending: {
    string: 'Sort from A to Z',
    number: 'Sort from 0 to 9'
  },
  descending: {
    string: 'Sort from Z to A',
    number: 'Sort from 9 to 0'
  }
};
var overrideStyles = null;

/**
 * __SortableColumn__
 *
 * SortableColumn is used in place of the default Column when sorting is desired.
 */
var SortableColumn = function SortableColumn(_ref) {
  var name = _ref.name,
    testId = _ref.testId,
    onClick = _ref.onClick,
    children = _ref.children,
    other = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _useTable = (0, _useTable2.useTable)(),
    sortKey = _useTable.sortKey,
    sortDirection = _useTable.sortDirection,
    setSortState = _useTable.setSortState;
  var getSortMessage = function getSortMessage() {
    if (sortKey === name) {
      // TODO: Change message depending on data type (string/number)
      return sortingMessages[sortDirection || 'unsorted'].string;
    }
    return sortingMessages.unsorted.string;
  };
  var updateSortState = (0, _react.useCallback)(
  // @ts-expect-error: TODO: Our `name` typing is off; refer to `SortKey`
  function () {
    return setSortState(name);
  }, [setSortState, name]);
  return /*#__PURE__*/React.createElement(_th.TH, (0, _extends2.default)({
    testId: testId,
    sortDirection: sortKey === name ? sortDirection : 'none'
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, other), /*#__PURE__*/React.createElement(_tooltip.default, {
    content: getSortMessage(),
    position: "top"
  }, function (tooltipProps) {
    return /*#__PURE__*/React.createElement(_button.default, (0, _extends2.default)({
      testId: "".concat(testId, "--button"),
      spacing: "compact",
      appearance: "subtle",
      iconAfter: /*#__PURE__*/React.createElement(_sortIcon.SortIcon, {
        name: name
      })
    }, tooltipProps, {
      onClick: updateSortState
      // TODO: (from codemod) Buttons with "css" or "style" prop can't be migrated for now. Please wait for the support of xcss prop.
      ,

      // TODO: i18n support for this attr
      "aria-roledescription": "Column sort button",
      className: (0, _runtime.ax)(["_19pkgrf3 _2hwxqyou _otyrgrf3 _18u0qyou _zulp1b66 _18zrud7t"])
    }), children);
  }));
};
var _default = exports.default = SortableColumn;
