// @atlaskit/table/dist/cjs/table.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _selectionProvider = _interopRequireDefault(require("./hooks/selection-provider"));
var _useSorting2 = require("./hooks/use-sorting");
var _useTable = require("./hooks/use-table");
var _ui = require("./ui");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
/**
 * __Table__
 *
 * A data table is used to display dynamic data.
 *
 * - [Examples](https://atlassian.design/components/table/examples)
 */
function Table(_ref) {
  var children = _ref.children,
    isSelectable = _ref.isSelectable,
    _ref$sortKey = _ref.sortKey,
    sortKey = _ref$sortKey === void 0 ? 'unset' : _ref$sortKey,
    testId = _ref.testId;
  var _useSorting = (0, _useSorting2.useSorting)(sortKey),
    localSortKey = _useSorting.sortKey,
    sortDirection = _useSorting.sortDirection,
    setSortState = _useSorting.setSortState,
    sortFn = _useSorting.sortFn;
  var tableProviderState = (0, _react.useMemo)(function () {
    return {
      isSelectable: isSelectable,
      sortKey: localSortKey,
      sortDirection: sortDirection,
      setSortState: setSortState,
      sortFn: sortFn
    };
  }, [isSelectable, localSortKey, setSortState, sortDirection, sortFn]);
  return /*#__PURE__*/_react.default.createElement(_useTable.TableProvider, {
    state: tableProviderState
  }, /*#__PURE__*/_react.default.createElement(_ui.Table, {
    testId: testId
  }, isSelectable ? /*#__PURE__*/_react.default.createElement(_selectionProvider.default, null, children) : children));
}
var _default = exports.default = Table;
