// @atlaskit/table/dist/cjs/hooks/use-table.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableProvider = TableProvider;
exports.useTable = void 0;
var _react = _interopRequireWildcard(require("react"));
var _noop = _interopRequireDefault(require("@atlaskit/ds-lib/noop"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
function generateContext() {
  return /*#__PURE__*/(0, _react.createContext)({
    isSelectable: false,
    sortKey: 'unset',
    setSortState: _noop.default,
    sortFn: function sortFn() {
      return 0;
    }
  });
}
var TableContext = generateContext();

/**
 * __Table state provider__
 *
 * The table context provides the data required for more complex functionality.
 *
 * - [Examples](https://atlassian.design/components/table/examples)
 */
function TableProvider(_ref) {
  var children = _ref.children,
    state = _ref.state;
  return (
    /*#__PURE__*/
    // @ts-expect-error
    _react.default.createElement(TableContext.Provider, {
      value: state
    }, children)
  );
}
var useTable = exports.useTable = function useTable() {
  return (
    // @ts-expect-error
    (0, _react.useContext)(TableContext)
  );
};
