// @atlaskit/table/dist/es2019/sortable-column.js
/* sortable-column.tsx generated by @compiled/babel-plugin v0.38.1 */
import _extends from "@babel/runtime/helpers/extends";
import "./sortable-column.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import { useCallback } from 'react';
import Button from '@atlaskit/button';
import Tooltip from '@atlaskit/tooltip';
import { useTable } from './hooks/use-table';
import { SortIcon as SortIconPrimitive } from './ui/sort-icon';
import { TH as THPrimitive } from './ui/th';
/**
 * TODO these need to be i18n supported
 */
const sortingMessages = {
  unsorted: {
    string: 'Sort from A to Z',
    number: 'Sort from 0 to 9'
  },
  ascending: {
    string: 'Sort from A to Z',
    number: 'Sort from 0 to 9'
  },
  descending: {
    string: 'Sort from Z to A',
    number: 'Sort from 9 to 0'
  }
};
const overrideStyles = null;

/**
 * __SortableColumn__
 *
 * SortableColumn is used in place of the default Column when sorting is desired.
 */
const SortableColumn = ({
  name,
  testId,
  onClick,
  children,
  ...other
}) => {
  const {
    sortKey,
    sortDirection,
    setSortState
  } = useTable();
  const getSortMessage = () => {
    if (sortKey === name) {
      // TODO: Change message depending on data type (string/number)
      return sortingMessages[sortDirection || 'unsorted'].string;
    }
    return sortingMessages.unsorted.string;
  };
  const updateSortState = useCallback(
  // @ts-expect-error: TODO: Our `name` typing is off; refer to `SortKey`
  () => setSortState(name), [setSortState, name]);
  return /*#__PURE__*/React.createElement(THPrimitive, _extends({
    testId: testId,
    sortDirection: sortKey === name ? sortDirection : 'none'
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, other), /*#__PURE__*/React.createElement(Tooltip, {
    content: getSortMessage(),
    position: "top"
  }, tooltipProps => /*#__PURE__*/React.createElement(Button, _extends({
    testId: `${testId}--button`,
    spacing: "compact",
    appearance: "subtle",
    iconAfter: /*#__PURE__*/React.createElement(SortIconPrimitive, {
      name: name
    })
  }, tooltipProps, {
    onClick: updateSortState
    // TODO: (from codemod) Buttons with "css" or "style" prop can't be migrated for now. Please wait for the support of xcss prop.
    ,

    // TODO: i18n support for this attr
    "aria-roledescription": "Column sort button",
    className: ax(["_19pkgrf3 _2hwxqyou _otyrgrf3 _18u0qyou _zulp1b66 _18zrud7t"])
  }), children)));
};
export default SortableColumn;
