// @atlaskit/table/dist/cjs/ui/bulk-action-overlay.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BulkActionOverlay = void 0;
var _react = _interopRequireDefault(require("react"));
var _primitives = require("@atlaskit/primitives");
// eslint-disable-next-line @atlaskit/design-system/no-emotion-primitives -- to be migrated to @atlaskit/primitives/compiled – go/akcss

var overlayStyles = (0, _primitives.xcss)({
  display: 'flex',
  position: 'absolute',
  inset: 'space.0',
  insetInlineStart: 'space.400'
});

/**
 * __Bulk action overlay__
 *
 * A bulk action overlay is used to conditionally render when a user makes a row selection
 */
var BulkActionOverlay = exports.BulkActionOverlay = function BulkActionOverlay(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    as: "th",
    paddingInline: "space.100",
    backgroundColor: "elevation.surface",
    xcss: overlayStyles
  }, /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    space: "space.300",
    alignBlock: "center"
  }, children));
};
