// @atlaskit/table/dist/cjs/ui/tr.js
/* tr.tsx generated by @compiled/babel-plugin v0.38.1 */
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TR = void 0;
require("./tr.compiled.css");
var React = _interopRequireWildcard(require("react"));
var _runtime = require("@compiled/react/runtime");
var _css = require("@atlaskit/css");
var _compiled = require("@atlaskit/primitives/compiled");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
var styles = {
  base: "_19itglyw _4t3i1wto _kqswh2mm _hpylidpf _btyzidpf",
  selected: "_bfhkcslv _irr3quvt",
  subitem: "_bfhkm7j4",
  bodyRow: "_1r62e4h9 _12ornqa1 _1ezz1tr0 _irr3134o"
};
/**
 * __Row__
 *
 * A row primitive.
 *
 * - [Examples](https://atlassian.design/components/table/examples)
 */
var TR = exports.TR = function TR(_ref) {
  var children = _ref.children,
    testId = _ref.testId,
    isSelected = _ref.isSelected,
    _ref$isBodyRow = _ref.isBodyRow,
    isBodyRow = _ref$isBodyRow === void 0 ? true : _ref$isBodyRow,
    isSubitem = _ref.isSubitem;
  return /*#__PURE__*/React.createElement(_compiled.Focusable, {
    as: "tr",
    isInset: true,
    tabIndex: -1,
    "aria-selected": isSelected,
    testId: testId,
    xcss: (0, _css.cx)(styles.base, isBodyRow && styles.bodyRow, isSelected && styles.selected, isSubitem && styles.subitem)
  }, children);
};
